src/rocksdb/src/config.rs

```diff
@@ -13,11 +13,229 @@
+    /// are updated in place.
+    pub fn apply(&mut self, params: RocksDBTuningParameters) {
+        let RocksDBTuningParameters {
+            compaction_style,
+            optimize_compaction_memtable_budget,
+            level_compaction_dynamic_level_bytes,
+            universal_compaction_target_ratio,
+            parallelism,
+            compression_type,
+            bottommost_compression_type,
+            batch_size,
+            retry_max_duration,
+            stats_log_interval_seconds,
+            stats_persist_interval_seconds,
+            point_lookup_block_cache_size_mb,
+            write_buffer_manager_memory_bytes,
+            write_buffer_manager_memory_fraction,
+            write_buffer_manager_allow_stall,
+        } = params;
+
+        self.compaction_style = compaction_style;
+        self.optimize_compaction_memtable_budget = optimize_compaction_memtable_budget;
+        self.level_compaction_dynamic_level_bytes = level_compaction_dynamic_level_bytes;
+        self.universal_compaction_target_ratio = universal_compaction_target_ratio;
+        self.parallelism = parallelism;
+        self.compression_type = compression_type;
+        self.bottommost_compression_type = bottommost_compression_type;
+        self.retry_max_duration = retry_max_duration;
+        self.stats_log_interval_seconds = stats_log_interval_seconds;
+        self.stats_persist_interval_seconds = stats_persist_interval_seconds;
+        self.point_lookup_block_cache_size_mb = point_lookup_block_cache_size_mb;
+
+        self.write_buffer_manager_config
+            .write_buffer_manager_memory_bytes = write_buffer_manager_memory_bytes;
+        self.write_buffer_manager_config
+            .write_buffer_manager_memory_fraction = write_buffer_manager_memory_fraction;
+        self.write_buffer_manager_config
+            .write_buffer_manager_allow_stall = write_buffer_manager_allow_stall;
+
+        // SeqCst is probably not required here, but its the easiest to reason about
+        self.dynamic.batch_size.store(batch_size, Ordering::SeqCst);
+    }
+}
+
+#[derive(Clone, Default, Derivative)]
+#[derivative(Debug)]
+pub struct SharedWriteBufferManager {
+    /// Keeping the shared [WriteBufferManager] behind an Arc and a Mutex.
+    /// Keeping a Weak pointer to WriteBufferManager here.
+    /// The strong pointers will be owned by each [`RocksDBInstance`].
+    /// When the rocksdb instances are cleaned up, the [WriteBufferManager] here will
+    /// be cleaned up as well.
+    #[derivative(Debug(format_with = "fmt_shared_write_buffer_manager"))]
+    shared: Arc<Mutex<Weak<WriteBufferManager>>>,
+}
+
+fn fmt_shared_write_buffer_manager(
+    buf: &Arc<Mutex<Weak<WriteBufferManager>>>,
+    fmt: &mut std::fmt::Formatter,
+) -> Result<(), std::fmt::Error> {
+    fmt.debug_struct("Weak<WriteBufferManager>")
+        .field(
+            "strong_count",
+            &buf.lock().expect("lock poisoned").strong_count(),
```

Comment:
Is this fine to log? I found it helpful because we do log the overall RocksDBConfig in upsert which would contain this.

Reply:
Ack, will remove

Reply:
done